src/index.js

```diff
@@ -0,0 +1,147 @@
+// const axios = require('axios');
+const state = {
+    city: 'Seattle',  
+    temperature: 20     
+};
+
+const getCurrentTemperature = () => {
+    const cityName = cityNameInput.value || 'Seattle';  
+
+    // Fetch latitude and longitude for the city
+    axios
+        .get('http://localhost:5000/location', {
+            params: { q: cityName }
+        })
+        .then(response => {
+            const { lat, lon } = response.data[0];  
+            console.log('Latitude:', lat);
+            console.log('Longitude:', lon);
+
+            return axios.get('http://localhost:5000/weather', {
+                params: { lat: lat, lon: lon }
+            });
+        })
+        .then(weatherResponse => {
+            const temperatureInKelvin = weatherResponse.data.main.temp;
+            const temperatureInFahrenheit = parseInt((temperatureInKelvin - 273.15) * 9 / 5 + 32);
+            console.log(`Temperature in Fahrenheit: ${temperatureInFahrenheit}°F`);
+
+            state.temperature = parseFloat(temperatureInFahrenheit, 10);  
+
+            tempValue.textContent = `${state.temperature}°F`;
+
+            headerCityName.textContent = cityName;
+
+            temperatureColor();
+            weatherGardenDisplay();
+        })
+        .catch(error => {
+            console.error('Error fetching weather data:', error);
+        });
+};
```

Comment:
I would suggest putting this logic for finding the temperature in its own function. The functionality you have its perfectly fine. It is more so for readability purposes. When it comes to promise chaining it can real nested/lengthy real quick. You could have make use of the `async/await` functionality that JS provides us too. Functionality will be the exact same, just easier to read.